### PR TITLE
Update UI and Improve Usability

### PR DESCRIPTION
**Pull request by niftynei**
A few small changes to the UI and adds some much needed navigation to messages in threads.  Includes updating to material design, so that the compose dialogs look more modern.  Also adds debug icons so that it's easy to tell the difference between a 'debug' build and a prod build, if you have them both loaded on your phone.

Incorporates changes requested in #34.

screenshot_20160326-105607
screenshot_20160326-105637
screenshot_20160326-105644
screenshot_20160326-105653
screenshot_20160326-105705
screenshot_20160326-105824
screenshot_20160326-105905

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@niftynei, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by timabbott** [comment]
The mockups look pretty good!  

A few things I noticed on looking at this (not reviewing the code carefully since I've done very little Android development):
* Looks like there's a small merge conflict with the .gitignore file from your other branch.
* In 6c4afce1a4a6a7539f9e5f653845dd9065e92126, why does it make sense to reverse the if statement in the R.id.search?  Probably makes sense to just put the reason in the commit message and refer back here.
* A similar note is that the "You and" thing I bet still looks funny if you PM yourself.
* One thought that I don't think needs to be resolved here but I was reminded of by the mocks is that it's somewhat problematic that there's no way to see the message you're replying to in the compose box.  Might be worth opening an issue for trying to fix that.

@lfaraone can you take a look?

**Comment by niftynei** [comment]
- the merge conflict with `.gitignore` has been resolved
- I added a longer explanatory note to the commit message; the short version is that for honeycomb+ two edit text boxes were being activated when you clicked search (a dialog & a field in the actionbar). only the actionbar edit text should show up.
- The "You and " PM _was_ a message to myself :) 
- Just opened issue #43 for this :+1:

File: app/src/main/java/com/zulip/android/ZulipActivity.java [hunk 1]
```diff
@@ -418,6 +418,7 @@ public void doNarrow(NarrowFilter filter) {
 
     @Override
     public void onNarrow(NarrowFilter narrowFilter) {
+        // TODO: check if already narrowed to this particular stream/subject
```
Comment (lfaraone) [inline app/src/main/java/com/zulip/android/ZulipActivity.java]: squash this into cbf90468

Comment (niftynei) [inline app/src/main/java/com/zulip/android/ZulipActivity.java]: is there a reason for squashing this?  it's a bit more general than just to that particular commit in that it applies to the current code AS-IS as well.

**Comment by niftynei** [comment]
@lfaraone bump. is there anything else I can change on this so we can get it merged in?

**Pull request merged**